### PR TITLE
[CI] [GHA] Use Dockerfiles in Linux CC workflow

### PR DESCRIPTION
**Pull request by akashchi**
### Tickets:
 - *140538*

### CODE REVIEW

File: .github/workflows/linux_conditional_compilation.yml [hunk 1]
```diff
@@ -40,17 +41,44 @@ jobs:
           component_pattern: "category: (.*)"
           repo_token: ${{ secrets.GITHUB_TOKEN }}
           skip_when_only_listed_labels_set: 'docs'
-          skip_when_only_listed_files_changed: '*.md,*.rst,*.png,*.jpg,*.svg,*/layer_tests_summary/*,*/conformance/*'
```
Comment (akladiev) [inline .github/workflows/linux_conditional_compilation.yml:43]: Was it removed intentionally?

File: .github/workflows/linux_conditional_compilation.yml [hunk 2]
```diff
@@ -72,17 +100,9 @@ jobs:
       SELECTIVE_BUILD_STAT_DIR: /__w/openvino/openvino/selective_build_stat
       MODELS_PATH: /__w/openvino/openvino/testdata
       SCCACHE_AZURE_KEY_PREFIX: ubuntu22_x86_64_itt_clang_Release
-    if: ${{ !needs.smart_ci.outputs.skip_workflow && github.event_name != 'merge_group' }}
+    if: ${{ !needs.Smart_CI.outputs.skip_workflow && github.event_name != 'merge_group' }}
```
Comment (akladiev) [inline .github/workflows/linux_conditional_compilation.yml:103]: Is it just for consistency? We usually kept it in lower case

File: .github/workflows/linux_conditional_compilation.yml (hunk 1 above)
Comment (akashchi) [reply .github/workflows/linux_conditional_compilation.yml:43]: unintentionally copied from `linux_arm64.yml`, reverted.

File: .github/workflows/linux_conditional_compilation.yml [hunk 3]
```diff
@@ -42,15 +43,42 @@ jobs:
...
+        uses: actions/checkout@v4
+
+      - uses: ./.github/actions/handle_docker
+        id: handle_docker
+        with:
+          images: |
+            ov_build/ubuntu_22_04_x64_cc
+          registry: 'openvinogithubactions.azurecr.io'
+          dockerfiles_root_dir: '.github/dockerfiles'
+          changed_components: ${{ needs.Smart_CI.outputs.changed_components }}
```
Comment (ilya-lavrenov) [inline .github/workflows/linux_conditional_compilation.yml:71]:
```suggestion
          changed_components: ${{ needs.smart_ci.outputs.changed_components }}
```

File: .github/workflows/linux_conditional_compilation.yml (hunk 2 above)
Comment (akashchi) [reply .github/workflows/linux_conditional_compilation.yml]: autoformatter did the thing, reverted.

**Review by akladiev — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**